python/pyspark/sql/connect/session.py

```diff
@@ -338,9 +342,31 @@ def catalog(self) -> "Catalog":
 
     catalog.__doc__ = PySparkSession.catalog.__doc__
 
+    def __del__(self) -> None:
```

Comment:
I piggyback this fix so garbage-collected instances will close the connection to avoid a resource leak.